shell/platform/darwin/ios/framework/Source/FlutterObservatoryPublisher.mm

```diff
@@ -113,11 +113,25 @@ - (void)publishServiceProtocolPort:(NSString*)uri {
 
   if (err != 0) {
     FML_LOG(ERROR) << "Failed to register observatory port with mDNS.";
+    if (@available(iOS 14.0, *)) {
+      FML_LOG(ERROR) << "Make sure the 'NSBonjourServices' key is set in your Info.plist for '"
+                     << registrationType << "' for the Debug configuration."
+                     << "(See also: https://developer.apple.com/news/?id=0oi77447)";
+    }
   } else {
     DNSServiceSetDispatchQueue(_dnsServiceRef, dispatch_get_main_queue());
   }
 }
 
+/// TODO(aaclarke): Remove this preprocessor macro once infra is moved to Xcode 12.
+static const DNSServiceErrorType kFlutter_DNSServiceErr_PolicyDenied =
+#if __IPHONE_OS_VERSION_MAX_ALLOWED >= 140000
```

Comment:
Can you use a `@available` here?  I've needed to use this macro outside of the class (like for imports) but hopefully `@available` works inline?  Not a big deal though.

Reply:
Na, that's why I messed it up the first time.  It's in an enum declaration that doesn't have annotations or guards of what version the operating system is.